utils.go

```diff
@@ -29,3 +36,10 @@ func DefaultIfEmpty(primary string, fallback string) string {
 	}
 	return primary
 }
+
+// IsTokenizerProxyError checks if the error comes from the Tokenizer.
+// It relies on the X-Proxy-Error header which will be set when it's TK internal error:
+// https://github.com/BoltApp/tokenization/blob/master/bolt/tk/app/handlers.ts#L24
+func IsTokenizerProxyError(header http.Header) bool {
+	return len(header.Get(HeaderXProxyError)) > 0
+}
```

Comment:
This is Bolt-specific logic that probably doesn't belong in sleet since non-Bolt clients of this library will not be proxying through Bolt TK. Can we move it into hail somehow?

Reply:
Added the HTTP response header into the `AuthorizationResponse`, so it can be parsed in hail.